crates/wasmtime/src/runtime/vm/table.rs

```diff
@@ -270,17 +270,22 @@ impl Table {
     pub fn new_dynamic(plan: &TablePlan, store: &mut dyn Store) -> Result<Self> {
         Self::limit_new(plan, store)?;
         let TableStyle::CallerChecksSignature { lazy_init } = plan.style;
+        let maximum = if !plan.table.table64 && plan.table.maximum.is_none() {
+            Some(usize::try_from(u32::MAX).unwrap())
+        } else {
+            plan.table.maximum.map(|i| usize::try_from(i).unwrap())
```

Comment:
Here this `.unwrap()` will want to be replaced with `.unwrap_or(usize::MAX)`. That'll mean that on a 32-bit system if you load a table with a maximum size of `u64::MAX` the effective maximum size of the table will end up being `usize::MAX`